contracts/mainnet/connectors/liquity/main.sol

```diff
@@ -196,37 +200,40 @@ abstract contract LiquityResolver is Events, Helpers {
      * @param repayAmount Amount of LUSD to repay
      * @param upperHint Address of the Trove near the upper bound of where the user's Trove should now sit in the ordered Trove list
      * @param lowerHint Address of the Trove near the lower bound of where the user's Trove should now sit in the ordered Trove list
-     * @param getDepositId Optional storage slot to retrieve the ETH to deposit
-     * @param setWithdrawId Optional storage slot to store the withdrawn ETH to
-     * @param getRepayId Optional storage slot to retrieve the LUSD to repay
-     * @param setBorrowId Optional storage slot to store the LUSD borrowed
+     * @param getIds Optional Get Ids for deposit, withdraw, borrow & repay
+     * @param setIds Optional Set Ids for deposit, withdraw, borrow & repay
     */
     function adjust(
         uint maxFeePercentage,
-        uint withdrawAmount,
         uint depositAmount,
+        uint withdrawAmount,
         uint borrowAmount,
         uint repayAmount,
         address upperHint,
         address lowerHint,
-        uint getDepositId,
-        uint setWithdrawId,
-        uint getRepayId,
-        uint setBorrowId
+        uint[] getIds,
+        uint[] setIds
     ) external payable returns (string memory _eventName, bytes memory _eventParam) {
-        if (getDepositId != 0 && depositAmount != 0) {
-            revert("adjust(): Cannot supply a depositAmount if a non-zero getDepositId is supplied");
-        }
-        if (getRepayId != 0 && repayAmount != 0) {
-            revert("adjust(): Cannot supply a repayAmount if a non-zero getRepayId is supplied");
-        }
         AdjustTrove memory adjustTrove;
 
         adjustTrove.maxFeePercentage = maxFeePercentage;
-        adjustTrove.withdrawAmount = withdrawAmount;
-        adjustTrove.depositAmount = getUint(getDepositId, depositAmount);
-        adjustTrove.borrowAmount = borrowAmount;
-        adjustTrove.repayAmount = getUint(getRepayId, repayAmount);
+
+        depositAmount = getUint(getIds[0], depositAmount);
+        adjustTrove.depositAmount = depositAmount == uint(-1) ? address(this).balance : depositAmount;
+
+        withdrawAmount = getUint(getIds[1], withdrawAmount);
+        adjustTrove.withdrawAmount = withdrawAmount == uint(-1) ? troveManager.getTroveColl(address(this)) : withdrawAmount;
+
+        adjustTrove.borrowAmount = getUint(getIds[2], borrowAmount);
```

Comment:
Do we need to check if the `borrowAmount` is `uint(-1)` here also? So we can use the DSA's LUSD balance

Reply:
How will we check `borrowAmount` as `uint(-1)` as borrow depends on user's position and not on user's LUSD balance. Basically, borrowing max can make position very risky and near to liquidation and will also add unnecessary logics.